appmap/flask.py

```diff
@@ -113,16 +117,15 @@ def before_request(self):
     def before_request_main(self, rec, request):
         Metadata.add_framework("flask", flask.__version__)
         np = None
-        # See
-        # https://github.com/pallets/werkzeug/blob/2.0.0/src/werkzeug/routing.py#L213
-        # for a description of parse_rule.
         if request.url_rule:
-            np = "".join(
-                [
-                    f"{{{p}}}" if c else p
-                    for c, _, p in parse_rule(request.url_rule.rule)
-                ]
+            # Transform requestl.url to the expected normalized-path form. For example,
+            # "/post/<username>/<post_id>/summary" becomes "/post/{{username}}/{{post_id}}/summary".
+            r = repr(request.url_rule)
+            np = NP_XLATE.sub(
+                lambda m: "{" if m["l"] else "}" if m["r"] else "",
```

Comment:
Should this be double `{{` and `}}` or is the comment inaccurate? Regardless, the comment could show the `repr` format so it's clear what's happening.

(I'm kind of weirded out by using `repr` like this, but as long as it works... also, let's hope rules never have a `'` in the name.)

Reply:
The comment is wrong. I'll add the `repr` format.

Using `repr` feels better to me than reaching into the implementation and using an undocumented interface (which was completely rewritten for Flask 2). The `repr` format is the same for both versions, so it should work for a least a little while....